enumerables.rb

```diff
@@ -0,0 +1,190 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size = count - 1
+    counter = 0
+    items = to_a
+    while counter <= size
+      if is_a? Range
+        items.length >= 0 && yield(items[counter])
+      elsif is_a? Hash
+        yield(items[counter][0], items[counter][1])
+      elsif !block_given?
+        return self
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+    self
+  end
+
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    counter = 0
+    size = length - 1
+
+    while counter <= size
+      yield(self[counter], counter)
+      counter += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    required = []
+
+    my_each do |r|
+      required.push(r) if yield(r)
+    end
+    required
+  end
+
+  def my_all?(*param)
+    track_negative = 0
+    size = count - 1
+    counter = 0
+
+    while counter <= size && !self[counter + 1].nil?
+      if param[0].class == Class
+        track_negative += self[counter].class == self[counter + 1].class ? 0 : 1
+        counter += 1
+      elsif param[0].class == Regexp
+        track_negative += self[counter].to_s.match(param[0]).nil? == self[counter + 1].to_s.match(param[0]).nil? ? 0 : 1
+        counter += 1
+      elsif block_given?
+        my_each do |p|
+          track_negative += yield(p) ? 0 : 1
+          counter += 1
+        end
+      elsif param[0].nil? && !block_given?
+        my_each do |ds|
+          track_negative += ds ? 0 : 1
+          counter += 1
+        end
+      end
+    end
+    !track_negative.positive?
+  end
+
+  def my_any?(*param)
+    yes = 0
+    size = count - 1
+    counter = 0
+
+    while counter <= size
+      if param[0].class == Class
+        yes += self[counter].class == param[0] ? 1 : 0
+        counter += 1
+      elsif param[0].class == Regexp
+        yes += self[counter].to_s.match(param[0]).nil? ? 0 : 1
+        counter += 1
+      elsif block_given?
+        my_each do |sth|
+          yes += yield(sth) ? 1 : 0
+          counter += 1
+        end
+      else
+        my_each do |sth|
+          yes += sth ? 1 : 0
+          counter += 1
+        end
+      end
+      yes
+    end
+    yes.positive?
+  end
+
+  def my_none?(*param)
+    satisfies = 0
+    size = count - 1
+    counter = 0
+
+    while counter <= size
+      if param[0].class == Class
+        satisfies += self[counter].class == param[0] ? 1 : 0
+        counter += 1
+      elsif param[0].class == Regexp
+        satisfies += self[counter].to_s.match(param[0]).nil? ? 0 : 1
+        counter += 1
+      elsif block_given?
+        my_each do |s|
+          satisfies += yield(s) ? 1 : 0
+          counter += 1
+        end
+      else
+        my_each do |t|
+          satisfies += t ? 1 : 0
+          counter += 1
+        end
+      end
+    end
+    !satisfies.positive?
+  end
```

Comment:
- For `my_none` when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
#### For example
`p ['cat','dog'].my_none?('fish') == false`
`p ['cat','dog'].none?('fish') == true`